implementations/rust/ockam/ockam_transport_tcp/src/workers/sender.rs

```diff
@@ -114,6 +118,30 @@ impl TcpSendWorker {
 
         Ok(())
     }
+
+    fn serialize_message(&self, local_message: LocalMessage) -> Result<Vec<u8>> {
+        // Create a message buffer with prepended length
+        let transport_message = TcpTransportMessage::from(local_message);
+
+        let msg_len = minicbor::len(&transport_message);
+
+        if msg_len > MAX_MESSAGE_SIZE {
+            return Err(TransportError::MessageLengthExceeded)?;
+        }
+
+        let len = 4 /* len */ + msg_len;
+
+        let msg_len_u32 =
+            u32::try_from(msg_len).map_err(|_| TransportError::MessageLengthExceeded)?;
+
+        let mut vec = vec![0u8; len];
+
+        vec[..4].copy_from_slice(&msg_len_u32.to_be_bytes());
+        minicbor::encode(&transport_message, &mut vec[4..])
+            .map_err(|_| TransportError::Encoding)?;
```

Comment:
Can we keep the encoding error?

Reply:
Do you think it would be useful? Decoding errors is what can happen and having details would be useful... encoding error are 99% bugs found during development